### DISCUSSION TITLE
lima doesn't restart the login session

### DISCUSSION THREAD

**Discussion by eminaktas** [lima-vm/lima#578] [Q&A]
I have tried to run docker with root, but adding docker group to the VM user doesn't apply (`sudo usermod -aG docker $USER`). It is like you have to start a new login session. You can do it by simply stopping and starting the VM. 

To see the behavior, you can use the root-docker.yaml example.

```bash
# Create the VM.
limactl start root-docker.yaml
# Login to VM.
limactl shell root-docker
```

When you run the command `limactl shell root-docker` with below example, you won't see the `docker` group for your user (default user is lima). However, you can check the groups for current user with `groups` command. Then run `sudo su - myUser`, now you will see the `docker` groups for your user. 

<details>

<summary>root-docker.yaml</summary>

```yaml
# Example to use Docker instead of containerd & nerdctl
# $ limactl start ./root-docker.yaml
# $ limactl shell docker docker run -it -v $HOME:$HOME --rm alpine

# To run `docker` on the host (assumes docker-cli is installed):
# $ export DOCKER_HOST=$(limactl list docker --format 'unix://{{.Dir}}/sock/docker.sock')
# $ docker ...

# This example requires Lima v0.8.0 or later
images:
# Hint: run `limactl prune` to invalidate the "current" cache
- location: "https://cloud-images.ubuntu.com/impish/current/impish-server-cloudimg-amd64.img"
  arch: "x86_64"
- location: "https://cloud-images.ubuntu.com/impish/current/impish-server-cloudimg-arm64.img"
  arch: "aarch64"
mounts:
- location: "~"
- location: "/tmp/lima"
  writable: true
# containerd is managed by Docker, not by Lima, so the values are set to false here.
containerd:
  system: false
  user: false
provision:
- mode: system
  script: |
    #!/bin/sh
    sed -i 's/host.lima.internal.*/host.lima.internal host.docker.internal/' /etc/hosts
- mode: system
  script: |
    #!/bin/bash
    set -eux -o pipefail
    command -v docker >/dev/null 2>&1 && exit 0
    export DEBIAN_FRONTEND=noninteractive
    curl -fsSL https://get.docker.com | sh
    # You can active it here but doesn't change the behavior.
    # usermod -aG docker lima # lima is default user for my system.
- mode: user
  script: |
    #!/bin/bash
    set -eux -o pipefail
    sudo usermod -aG docker $USER
portForwards:
- guestSocket: "/var/run/docker.sock"
  hostSocket: "{{.Dir}}/sock/docker.sock"
message: |
  To run `docker` on the host (assumes docker-cli is installed):
  $ export DOCKER_HOST=unix://{{.Dir}}/sock/docker.sock
  $ docker ...
```
</details>

Basically this causes the Docker socket is not usable from the host machine, but inside VM, it is usable when you run `sudo su - myUser` or root user.

```
error during connect: Get "http://%2FUsers%2FMACPC%2F.lima%2Fdocker%2Fsock%2Fdocker.sock/v1.24/containers/json": EOF
```

I am not sure, but this shouldn't be the behavior. Whenever I start a new VM, all settings and commands in the YAML file should be usable. 
Also, there is a discussion [here](https://github.com/kubernetes/minikube/issues/12508).

**Comment by afbjorklund** [comment]
Also mentioned here:
* #528
